backend/common/repo.py

```diff
@@ -110,11 +110,17 @@ def get_pkg_index_raw(self) -> typing.Tuple[str, bytes]:
         if self._pkg_index[0] == "":
             for cnt_fname in [DpkgRepo.PKG_GZ, DpkgRepo.PKG_XZ, DpkgRepo.PKG_RW]:
                 packages_url = self.append_index_file(cnt_fname)
-                resp = requests.get(packages_url, proxies=self.proxies)
-                if resp.status_code == http.HTTPStatus.OK:
-                    self._pkg_index = cnt_fname, resp.content
-                    break
-                resp.close()
+                if packages_url.startswith("file://"):
+                    with open(packages_url.replace("file://", ""), "rb") as f:
+                        self._pkg_index = cnt_fname, f.read()
+                        break
+                    # TODO: Add logging in error case!
```

Comment:
Could we do the actual logging instead of having this `TODO` here?

Reply:
There is no logging at all. I'd rather do a second PR to introduce logging. Actually we need logging for several more cases here.